services/cd-service/pkg/repository/transformer.go

```diff
@@ -588,9 +589,32 @@ type CreateEnvironmentLock struct {
 	Environment string
 	LockId      string
 	Message     string
+	RBACConfig  auth.RBACConfig
```

Comment:
I see two options here:
1) Add `RBACConfig` as another parameter to the transformer (and remove it here).
2) Or you separate the RBACConfig with embedding:
```
type RBACConfig struct {
	/// ...
}

type CreateEnvironmentLock struct {
	RBAC // embed
	Environment string
	LockId      string
	Message     string
}
```
Either way, there should be some separation between general parameters and the RBAC data.